src/Tokens/Models/UserIdToken/UserIdToken.php

```diff
@@ -162,6 +175,11 @@ private function parseToken(): array
             throw new InvalidUserIdTokenException(sprintf('Unable to parse JSON string %s.', $payloadJsonString));
         }
 
-        return (array)$payload;
+        return is_array($payload) ? $payload : (array)$payload;
+    }
+
+    private function convertTimestampToDateTime(int $timeStamp): DateTimeImmutable
+    {
+        return (new DateTimeImmutable())->setTimestamp($timeStamp);
```

Comment:
You can actually initialize directly from a timestamp by doing: `new DateTimeImmutable('@' . $timeStamp);`